### PR TITLE
fix($location): Infinite digest in $location replace in IE9

### PR DESCRIPTION
**Pull request by petebacondarwin**
@hamfastgamgee - I think this is the right unit test. Can you take a look. The nice thing is that it even fails on Chrome as well (since we are simulating no-history support) without the fix.

### CODE REVIEW

**Comment by hamfastgamgee** [comment]
Well, this is certainly a much more concise version of what I was directly fixing.  I'm going to update my patch in the other pull request with something I was knocking around this morning that I'd like you to ponder.  I believe the problem is somewhat more widespread than previously thought (impacting more cases) because the mock browser didn't expose the bug.

**Comment by petebacondarwin** [comment]
Perhaps we do need to update ngMock browser across the board as you say?

**Comment by hamfastgamgee** [comment]
Seems at the least like it would be a more honest accounting of what's really going wrong in the app, which should lead to more bullet-proof fixes. :)

**Comment by hamfastgamgee** [comment]
Is there an easy way to run the karma tests on another browser locally?  (I'm relatively new to grunt.)

**Comment by petebacondarwin** [comment]
Easiest way to run tests locally is to use `grunt autotest`. This will setup karma to watch the source and test files and automatically run the tests on any change. Once this is running, you can then attach additional browsers by navigating them to http://localhost:9876

**Comment by hamfastgamgee** [comment]
Thanks!  Please see new version over on #11675.

**Comment by hamfastgamgee** [comment]
So is this fixed, or...?

**Comment by petebacondarwin** [comment]
Moving focus to #11675. I will try to land this in the next couple of days :-)

**Comment by hamfastgamgee** [comment]
Gotcha. Thanks again for all your help! 
On Jun 1, 2015 7:04 AM, Pete Bacon Darwin <notifications@github.com> wrote:Moving focus to #11675. I will try to land this in the next couple of days :-)

File: test/ng/locationSpec.js [hunk 1]
```diff
@@ -996,6 +996,49 @@ describe('$location', function() {
...
+
+    it('should not get caught in infinite digest when replacing empty path with slash', function() {
+      initService({html5Mode:true,supportHistory:false});
+      mockUpBrowser({initialUrl:'http://server/base', baseHref:'/base/'});
+      inject(
+        function($browser, $location, $rootScope, $window) {
+          $rootScope.$on('$locationChangeSuccess', function() {
+            if ($location.path() !== '/') {
+                $location.path('/').replace();
+            }
```
Comment (IgorMinar) [inline test/ng/locationSpec.js]: add an assertion here

File: test/ng/locationSpec.js [hunk 2]
```diff
@@ -996,6 +996,49 @@ describe('$location', function() {
...
+    }
+
+
+    it('should not get caught in infinite digest when replacing empty path with slash', function() {
+      initService({html5Mode:true,supportHistory:false});
+      mockUpBrowser({initialUrl:'http://server/base', baseHref:'/base/'});
+      inject(
+        function($browser, $location, $rootScope, $window) {
+          $rootScope.$on('$locationChangeSuccess', function() {
+            if ($location.path() !== '/') {
```
Comment (petebacondarwin) [inline test/ng/locationSpec.js]: need to assert that this handler was actually called

File: test/ng/locationSpec.js [hunk 3]
```diff
@@ -996,6 +996,49 @@ describe('$location', function() {
...
+                $location.path('/').replace();
+            }
+          });
+          $rootScope.$digest();
+        }
```
Comment (petebacondarwin) [inline test/ng/locationSpec.js]: Need to check that the $location actually got updated after the digest

Comment (IgorMinar) [inline test/ng/locationSpec.js]: add an assertion that we executed the locationChangeSuccess callback

**Pull request closed without merging**